report/verification/mms/simple_transient.md

```diff
@@ -0,0 +1,318 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.16.2
+kernelspec:
+  display_name: vv-festim-report-env
+  language: python
+  name: python3
+---
+
+# Simple transient diffusion case
+
+```{tags} 2D, MMS, transient
+```
+
+This is a simple transient MMS example.
+We will only consider diffusion of hydrogen in a unit square domain $\Omega$ at steady state with an homogeneous diffusion coefficient $D$.
+Moreover, a Dirichlet boundary condition will be assumed on the boundaries $\partial \Omega $.
+
+The problem is therefore:
+
+$$
+\begin{align}
+    &\nabla \cdot (D \ \nabla{c}) - \frac{\partial c}{\partial t} = -S  \quad \text{on }  \Omega  ; \ t\geq 0 \\
+    & c = c_0 \quad \text{on }  \partial \Omega ; \ t\geq 0 \\
+    & c = c_\mathrm{initial} \quad \text{at } t=0
+\end{align}
+$$(problem_simple_transient)
+
+The exact solution for mobile concentration is:
+
+$$
+\begin{equation}
+    c_\mathrm{exact} = 1 + 2 x^2 + 3 y^2 t + 2t
+\end{equation}
+$$(c_exact_simple_transient)
+
+Injecting {eq}`c_exact_simple_transient` in {eq}`problem_simple_transient`, we obtain the expressions of $S$ and $c_0$:
+
+\begin{align}
+    & S = 2 + 3 y^2 - (4 + 6t) D \\
+    & c_0 = c_\mathrm{exact} \\
+    & c_\mathrm{initial} = c_\mathrm{exact}(t=0)
+\end{align}
+
+We can then run a FESTIM model with these values and compare the numerical solution with $c_\mathrm{exact}$.
+
++++
+
+## FESTIM code
+
+```{code-cell} ipython3
+:tags: [hide-cell]
+
+import festim as F
+import sympy as sp
+import fenics as f
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import numpy as np
+
+# Create and mark the mesh
+nx = ny = 100
+fenics_mesh = f.UnitSquareMesh(nx, ny)
+
+
+volume_markers = f.MeshFunction("size_t", fenics_mesh, fenics_mesh.topology().dim())
+volume_markers.set_all(1)
+
+surface_markers = f.MeshFunction(
+    "size_t", fenics_mesh, fenics_mesh.topology().dim() - 1
+)
+surface_markers.set_all(0)
+
+class Boundary(f.SubDomain):
+    def inside(self, x, on_boundary):
+        return on_boundary
+
+boundary = Boundary()
+boundary.mark(surface_markers, 1)
```

Comment:
I think we could make this a function that takes `n` as argument, and returns `fenics_mesh`, `volume_markers` and `surface_markers`